### PR TITLE
Move programme observation

### PR DESCRIPTION
**Pull request by urbanwide**
Move observations for programmes to use the new polymorphic relationship.
Keeps the old programme_id column in the table for now, will remove all these when they are all moved.

Worth noting that I don't think there are any observations of these types in the database - has nobody ever completed a programme? Or maybe it's not working (there are tests which indicate the code should be working)?

### CODE REVIEW

File: db/migrate/20231207104401_move_programme_observation.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class MoveProgrammeObservation < ActiveRecord::Migration[6.0]
+  def up
+    Observation.where(observation_type: :programme).find_each do |observation|
+      observation.obserable = observation.programme
```
Comment (ldodds) [inline db/migrate/20231207104401_move_programme_observation.rb:4]: There's a typo here I think?

**Review by ldodds — CHANGES_REQUESTED** [review]
Looks good apart from the migration

**Comment by ldodds** [comment]
There's one Programme with a `:completed` status, so there should be an observation for it?

File: db/migrate/20231207104401_move_programme_observation.rb (hunk 1 above)
Comment (urbanwide) [reply db/migrate/20231207104401_move_programme_observation.rb:4]: Oh gosh, good spot, thank you. Strange that it ran (or perhaps I changed it afterwards).

**Comment by urbanwide** [comment]
I couldn't see one - however it looks like the programme observation feature was added this July, after the programme was completed. I couldn't find an after party task to add retrospective programme observations in, want me to do this?

**Comment by ldodds** [comment]
This was obviously missed! If you can add a task for that as a separate PR, that would be great. Would also then ensure your migration gets exercised

**Review by ldodds — APPROVED** [review]

**Pull request merged**